Sources/UIKit/TextAlignmentConstraint.swift

```diff
@@ -0,0 +1,162 @@
+//
+//  TextAlignmentConstraint.swift
+//
+//  Created by Cameron Pulsford on 10/4/16.
+//  Copyright © 2016 Raizlabs. All rights reserved.
+//
+
+#if os(OSX)
+    import AppKit
+#else
+    import UIKit
+#endif
+
+@objc(BONTextAlignmentConstraintAttribute)
+public enum TextAlignmentConstraintAttribute: Int {
+    case unspecified
+    case top
+    case capHeight
+    case xHeight
+    case firstBaseline
+    case lastBaseline
+    case bottom
+}
+
+private extension TextAlignmentConstraintAttribute {
+    var layoutAttribute: NSLayoutAttribute {
+        switch self {
+        case .top, .capHeight, .firstBaseline, .xHeight:
+            return .top
+        case .lastBaseline, .bottom:
+            return .bottom
+        case .unspecified:
+            return .notAnAttribute
+        }
+    }
+}
+
+private var TextAlignmentConstraintKVOContext = "TextAlignmentConstraintKVOContext" as NSString
```

Comment:
Might want to add BonMot to the string, to make debugging easier and provide even better defense against collisions. Also, might want to embed this inside `TextAlignmentConstraint`